ols/src/llms/llm_loader.py

```diff
@@ -49,7 +49,7 @@ def _resolve_provider_config(
             f"Valid models are: {list(provider_config.models.keys())}"
         )
 
-    return provider_config
+    return provider_config  # type: ignore [return-value]
```

Comment:
why do we need to ignore the type?  is ` providers_config.providers.get(provider)` not setting a return type?

basically same question on all of these.  Why do we have to ignore the return type? doesn't that imply we're not setting types correctly somewhere else?

Reply:
providers is:
```
providers: dict[str, ProviderConfig] = {}
```

so why would providers.get() not be expected to return an object of type ProviderConfig? (or perhaps `ProviderConfig | None`)?

Reply:
`ProviderConfig | None` is the problem there, because `get` might return `None`. So either the get will be replaced by direct selector or we ignore return value type.

Reply:
or we change this:
```
def _resolve_provider_config(
    provider: str, model: str, providers_config: LLMProviders
) -> ProviderConfig:
```

to
```
def _resolve_provider_config(
    provider: str, model: str, providers_config: LLMProviders
) -> ProviderConfig|None:
```

(or whatever the syntax is), right?

(even though we never really return none because we throw an exception if providerconfig is none)

Reply:
yes, but transitionally it will "force" us to check for None in every piece of code that use provider.

Reply:
got it

Reply:
i am increasingly getting the impression that exhaustive type checking is going to be:

1) difficult to implement
2) difficult to maintain
3) have significant limitations such as this one that will compromise the value it can provide or create confusion about when/why it's ok to ignore types

Reply:
fair enough